packages/components/src/card/index.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import classnames from 'classnames';
+import Gridicon from 'gridicons'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type TagName< P = any > = React.ElementType< P >; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+type OwnProps = {
+	className?: string;
+	displayAsLink?: boolean;
+	href?: string;
+	target?: string;
+	compact?: boolean;
+	highlight?: 'error' | 'info' | 'success' | 'warning';
+};
+
+type ElementProps< P, T extends TagName > = P &
+	Omit< React.ComponentProps< T >, 'tagName' | keyof P > & {
+		tagName?: T | keyof JSX.IntrinsicElements;
+		children?: React.ReactNode;
+	};
```

Comment:
This is taken from my work in G2 and basically mirrors what's done in Reakit for similar polymorphic components.

The gist of what's happening here is that we create a type that is a combination of:
1. The "regular" props (P)
2. The props for the component type (`React.ComponentProps<T>`) less the props that overlap with `tagName` or `P`
3. `tagName` and `children`.

Reply:
@sirreal any ideas on this one?

Reply:
👋  What's the question? I checked this out and it seems to compile.

Reply:
Sorry I thought I had put the questions above but it seems that I did not.

Basically `Card` has the same behavior as `Button` where if you pass `href` it turns it into a an anchor element rather than respecting what you passed into `tagName`. Should I just try to replicate what was done for `Button` or... is there a different way you can see this working?